### PR TITLE
Add example for Go with Docker

### PR DESCRIPTION
**Pull request by tasiah**
For pulumi/pulumi-docker#128. Relies on pulumi/pulumi-docker#147.

### CODE REVIEW

File: azure-go-aci/main.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		portArgs := containerservice.GroupContainerPortArgs{
+			Port:     pulumi.Int(80),
+			Protocol: pulumi.String("TCP"),
+		}
+		containerArgs := containerservice.GroupContainerArgs{
+			Cpu:    pulumi.Float64(0.5),
+			Image:  image.ImageName,
+			Memory: pulumi.Float64(1.5),
+			Name:   pulumi.String("hello-world"),
+			Ports:  containerservice.GroupContainerPortArray([]containerservice.GroupContainerPortInput{portArgs}),
```
Comment (lukehoban) [inline azure-go-aci/main.go]: @pgavlin has pointed out to me that in general - this pattern can always be written as this instead:

```go
containerservice.GroupContainerPortArray({portArgs}),
```

Same applies in a few more places in this file (and in other Go examples we have).

Comment (tasiah) [reply azure-go-aci/main.go]: When I try `containerservice.GroupContainerPortArray({portArgs})`, Go says it's a syntax error: `syntax error: unexpected {, expecting expression`. Am I missing something?

Comment (lukehoban) [reply azure-go-aci/main.go]: Sorry - typed that wrong - it's:

```go
containerservice.GroupContainerPortArray{portArgs},
```

**Comment by tasiah** [comment]
Should I wait to merge this until the next release of pulumi/pulumi-docker?

**Comment by tasiah** [comment]
I realized I could just update the `Gopkg.toml` to use the latest revision (SHA) of `pulumi/pulumi-docker`. So after pulumi/pulumi-docker#149 gets merged, I'll update the `Gopkg.toml` to that SHA and merge this example (finally!).

**Pull request merged**